radicle-httpd/src/api/v1/projects.rs

```diff
@@ -236,10 +253,30 @@ async fn tree_handler(
     let path = path.strip_prefix('/').ok_or(Error::NotFound)?.to_string();
     let storage = &ctx.profile.storage;
     let repo = Repository::open(paths::repository(storage, &project))?;
-    let path = if path.is_empty() { None } else { Some(&path) };
-    let tree = Tree::new(&repo, &sha, path)?;
-    let mut response = json!(tree);
-    response["stats"] = json!(stats(&repo, sha)?);
+    let tree = repo.tree(sha, &path)?;
+
+    let prefix = std::path::Path::new(&path);
+    let entries = tree
+        .entries()
+        .iter()
+        .map(|entry| {
+            json!( {
+                "path": prefix.join(entry.name()),
+                "name": entry.name(),
+                "lastCommit": serde_json::Value::Null,
+                "kind": if entry.is_tree() { "tree" } else { "blob" },
+            })
+        })
+        .collect::<Vec<_>>();
+
+    let last_commit = tree.commit();
+    let response = json!({
+        "entries": &entries,
+        "lastCommit": commit_json(last_commit),
+        "name": name_in_path(&path),
+        "path": &path,
+        "stats": repo.stats()?,
```

Comment:
This looks incorrect; what commit will it use for the stats?

Reply:
Ohhhh, was there a conversation about this before... I think you want to use `stats_from` here https://github.com/radicle-dev/radicle-git/blob/main/radicle-surf/src/repo.rs#L334-L353

Reply:
Yeah there was :)

Reply:
I missed that, thanks!